src/main/java/calculator/core/service/CalculateService.java

```diff
@@ -0,0 +1,51 @@
+package calculator.core.service;
+
+import calculator.converter.StringToPositiveBigIntegerConverter;
+import calculator.extractor.TextCustomDelimiterExtractor;
+import calculator.splitter.TextDelimiterSplitter;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Set;
+
+import static calculator.delimiter.BasicDelimiter.getBasicDelimiters;
+import static calculator.delimiter.CustomDelimiter.getCustomDelimiters;
+import static calculator.delimiter.CustomDelimiter.hasCustomDelimiter;
+import static calculator.util.StringUtils.hasText;
+import static java.math.BigInteger.ZERO;
+
+public class CalculateService {
+
+    private final TextDelimiterSplitter textDelimiterSplitter;
+    private final TextCustomDelimiterExtractor textCustomDelimiterExtractor;
+    private final StringToPositiveBigIntegerConverter stringToPositiveBigIntegerConverter;
+
+    public CalculateService() {
+        this.textDelimiterSplitter = new TextDelimiterSplitter();
+        this.textCustomDelimiterExtractor = new TextCustomDelimiterExtractor();
+        this.stringToPositiveBigIntegerConverter = new StringToPositiveBigIntegerConverter();
+    }
+
+    public BigInteger calculate(String input) {
+        if (!hasText(input)) {
+            return ZERO;
+        }
+
+        if (!hasCustomDelimiter(input)) {
+            return sum(textDelimiterSplitter.split(input, getBasicDelimiters()));
+        }
```

Comment:
부정문을 사용하기보다 긍정문을 사용하라고 합니다 !
```suggestion
        if (hasNotText(input)) {
            return ZERO;
        }

        if (hasNotCustomDelimiter(input)) {
            return sum(textDelimiterSplitter.split(input, getBasicDelimiters()));
        }
```
이런식으로 하시고 내부에서 '!'를 이용하시는게 가독성에 더 좋다고해요!

Reply:
아하 넵 가독성을 위해서라면! 🙇‍♂️